#cob4-2
##########
# common params
run_factor: 2.0
joy_num_buttons: 12
joy_num_axes: 6
joy_num_modes: 7
mode_switch_button: 0

# axes
axis_vx: 17
axis_vy: 16
axis_vz: 5 #buttons are not axis 4-6
axis_roll: 4 #buttons are not axis 7-5
axis_pitch: 3
axis_yaw: 19

# buttons
deadman_button: 11
safety_button: 10
init_button: 3

#mode1: Base
run_button: 7 #is it a common parameter? should we use this button also for other twist controllers or velocity controllers?

#mode2: Trajectory controller
torso_default_target_button: 6
sring_default_target_button: 12
head_default_target_button: 4
arm_right_default_target_button: 5
arm_left_default_target_button: 7

#mode3: Velocity group controller
torso_joint1_button: 15
sensorring_joint1_button: 14
head_joint1_button: 13
arm_right_joint1_button: 15
arm_right_joint2_button: 14
arm_right_joint3_button: 13
arm_right_joint4_button: 12
arm_left_joint1_button: 15
arm_left_joint2_button: 14
arm_left_joint3_button: 13
arm_left_joint4_button: 12

right_indicator_button: 9
left_indicator_button: 8
up_down_button: 5
right_left_button: 4

#mode4: Twist controller
torso_twist_button: 14
head_twist_button: 12
arm_right_twist_button: 13
arm_left_twist_button: 15

components: {
  base: {
    twist_topic_name: '/base/twist_controller/command',
    max_velocity: [0.3, 0.2, 0.3],
    max_acceleration: [0.5, 0.5, 0.7]
  },
  torso: {
    twist_topic_name: '/torso/twist_controller/command',
    topic_name: "/torso/joint_group_velocity_controller/command",
    default_target: "home",
    max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3],
    joint_vel: [0.05,0.05]
  },
  head: {
    twist_topic_name: '/head/twist_controller/command',
    topic_name: "/head/joint_group_velocity_controller/command",
    default_target: "front",
    max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3],
    joint_vel: [0.05,0.05]
  },
  sensorring: {
    twist_topic_name: '/sensorring/twist_controller/command',
    topic_name: "/sensorring/joint_group_velocity_controller/command",
    default_target: "front",
    joint_vel: [0.05]
  },
  arm_left: {
    twist_topic_name: '/arm_left/twist_controller/command',
    topic_name: "/arm_left/joint_group_velocity_controller/command",
    default_target: "home",
    max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3],
    joint_vel: [0.05,0.05,0.05,0.05,0.05,0.05,0.05]
  },
  arm_right: {
    twist_topic_name: '/arm_right/twist_controller/command',
    topic_name: "/arm_right/joint_group_velocity_controller/command",
    default_target: "home",
    max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3],
    joint_vel: [0.05,0.05,0.05,0.05,0.05,0.05,0.05]
  }
}
